packages/medusa/src/services/currency.ts

```diff
@@ -9,48 +8,54 @@ import { UpdateCurrencyInput } from "../types/currency"
 import { buildQuery } from "../utils"
 import { FlagRouter } from "../utils/flag-router"
 import EventBusService from "./event-bus"
+import { DbTransactionService } from "./index"
 
 type InjectedDependencies = {
   manager: EntityManager
   currencyRepository: typeof CurrencyRepository
   eventBusService: EventBusService
   featureFlagRouter: FlagRouter
+  dbTransactionService: DbTransactionService
 }
 
-export default class CurrencyService extends TransactionBaseService {
+export default class CurrencyService {
   static readonly Events = {
     UPDATED: "currency.updated",
   }
 
   protected manager_: EntityManager
-  protected transactionManager_: EntityManager | undefined
 
   protected readonly currencyRepository_: typeof CurrencyRepository
   protected readonly eventBusService_: EventBusService
   protected readonly featureFlagRouter_: FlagRouter
+  protected readonly dbTransactionService_: DbTransactionService
 
   constructor({
     manager,
     currencyRepository,
     eventBusService,
     featureFlagRouter,
+    dbTransactionService,
   }: InjectedDependencies) {
-    super(arguments[0])
     this.manager_ = manager
     this.currencyRepository_ = currencyRepository
     this.eventBusService_ = eventBusService
     this.featureFlagRouter_ = featureFlagRouter
+    this.dbTransactionService_ = dbTransactionService
   }
 
   /**
    * Return the currency
-   * @param code - The code of the currency that must be retrieve
+   * @param code - The code of the currency that must be retrieved
+   * @param context
    * @return The currency
    */
-  async retrieveByCode(code: string): Promise<Currency | never> {
-    const currencyRepo = this.manager_.getCustomRepository(
-      this.currencyRepository_
-    )
+  async retrieveByCode(
+    code: string,
+    { transactionManager }: { transactionManager?: EntityManager } = {}
```

Comment:
suggestion: Should we define a type for context with transactionManager? I'm a bit concerned about potential typos. Could also help with future refactors.

Reply:
Yes I agree, what about TransactionContext and if a context includes more than just the transaction manager then the person implementing it can just do a union type of what is expected + TransactionContext